packages/sqlite/src/lib/LibSqlClient.ts

```diff
@@ -59,6 +59,42 @@ export class LibSqlWriter {
 
     return { type: "Written", position }
   }
+
+  async writeSnapshot(
+    category: string,
+    streamName: string,
+    event: IEventData<Format>,
+    expectedVersion?: bigint,
+  ): Promise<LibSqlWriteResult> {
+    const trx = await this.client.transaction("write")
+    try {
+      const current = await trx.execute({
+        sql: "select position from snapshots where stream_name = ?",
+        args: [streamName],
+      })
+      const currentVersion = current.rows.length === 0 ? -1n : BigInt(current.rows[0][0] as number)
```

Comment:
as noted earlier, a version on its own is not great for protecting in the face of multiple writers 
or maybe it is ;) just making sure you have some concurrency model in mind.

Reply:
I guess there's little point in ignoring the fundamentals of the backing store
But please tell me the version starts at 0 when there's no events!
A one liner at top of file might be good - want people scanning stores to understand that in general the version is the event count
and that rollingstate or custom means that there is an etag that further dictates whether you get to write just because you know the version
Also if you bake in the rolling state mode as the version in this way, you preclude ever having events - the Custom mode allows you to e.g. produce a checkpoint every 1000 updates or every hour etc

----

TL;DR if it was me, I think I'd have events be marked with a zero-based index like the rest of the world

and then in the snapshot table keep a tinyint etag thing that increments with wraparound

having said that, if not doing that means you can simplify, of course it's worth considering (but in my head I want to be able to easily move from defaulting to rollingsnapshot, and then deiding I want actual events for some things, without the version suddenly meaning something different entirely

Still 🤔 though